lorawan-device/src/mac/session.rs

```diff
@@ -314,21 +317,20 @@ impl Session {
         configuration: &mut super::Configuration,
         region: &mut region::Configuration,
         cmds: MacCommandIterator<'_, DownlinkMacCommand<'_>>,
+        snr: i8,
     ) {
         use DownlinkMacCommand::*;
         let mut channel_mask = region.channel_mask_get();
         let mut cmd_iter = cmds.into_iter().peekable();
         let mut num_adrreq = 0;
         while let Some(cmd) = cmd_iter.next() {
             match cmd {
-                #[cfg(feature = "experimental")]
                 DevStatusReq(..) => {
-                    // TODO: Fill with proper values
-                    // - Battery: (255 - unable to measure, 1..254 - battery level, 0 - external power source)
-                    // - RadioStatus: (SNR: -32..31)
-                    // For now we just return dummy values
+                    // TODO: Battery information should come from device/application
+                    // Battery: (255 - unable to measure, 1..254 - battery level, 0 - external power source)
+                    // For now we just return dummy value of "255"
                     let mut cmd = DevStatusAnsCreator::new();
-                    let _ = cmd.set_battery(255).set_margin(0);
+                    let _ = cmd.set_battery(255).set_margin(snr);
```

Comment:
Do we have any guarantees that the snr is 6-bit here? Seems to me like we get whatever the radio is returning, there is a cast from i16 to i8 here

Reply:
https://github.com/lora-rs/lora-rs/blob/9223e421135ebf95af0f67a07ee33715c77465c7/lorawan-encoding/src/maccommandcreator.rs#L361-L368

Ok so there are some checks here but without using the result we probably end up with 0 here if the value is wrong. Idk if it ever would be a problem.

Reply:
Yeah, I actually had these concerns as well, but I didn't want to go down the rabbit hole too deeply.
Ideally there should be a proper LoRaWAN "margin" type instead of i8.

Reply:
true, should not be that big of a foot gun at this stage so good idea to not bloat the PR. All we get is bs answers but at least we get an answer. This command is used a lot by thingpark at least. Idk if it affects ADR but they like to show the battery status.